scripts/private/sparkk8s_token.sh

```diff
@@ -19,69 +19,66 @@ fi
 USERNAME="$1"
 SERVICE_ACCOUNT="spark"
 
-KUBECONFIG="/srv/jupyterhub/private/sparkk8s.cred"
+export KUBECONFIG="/srv/jupyterhub/private/sparkk8s.cred"
 
 if [[ ! -f $KUBECONFIG ]]; then
     exit 1
 fi
 
-if [ "$SWAN_DEV" = "true" ]; then
-    # For dev purposes, one can provide already generated token
-    echo $(cat $KUBECONFIG | base64 -w 0)
-    exit 0
-fi
-
 SERVER=$(awk -F"server: " '{print $2}' ${KUBECONFIG} | sed '/^$/d')
 
-helm init --client-only > /dev/null 2>&1
-
-if user_exists=$(helm --kubeconfig "${KUBECONFIG}" ls "spark-user-${USERNAME}" > /dev/null 2>&1); then
-    :
-else
-    echo "ERROR: cluster is not ready. Please initialize cluster with helm manually"
-    exit 1
-fi
+# Check if the helm chart for the user is already installed
+user_exists=$(helm --kubeconfig "${KUBECONFIG}" list -n ${USERNAME} --filter "spark-user-${USERNAME}" 2>/dev/null | grep -v NAMESPACE)
 
-if [[ -z "${user_exists// }" ]]; then
+# Install the helm chart for the user if it does not exist
+if [[ -z "${user_exists}" ]]; then
     # User not initialized
     helm install \
-    --wait \
+    spark-user-${USERNAME} \
+    https://gitlab.cern.ch/db/spark-service/spark-service-charts/-/raw/v104/cern-spark-user-1.0.4.tgz \
     --kubeconfig "${KUBECONFIG}" \
-    --set namespace="${USERNAME}" \
+    --namespace ${USERNAME} \
+    --create-namespace \
+    --set namespace=${USERNAME} \
     --set cvmfs.enable=true \
-    --name "spark-user-${USERNAME}" https://gitlab.cern.ch/db/spark-service/spark-service-charts/raw/master/cern-spark-user-1.0.2.tgz > /dev/null 2>&1
+    --wait > /dev/null 2>&1
 fi
 
-# Retrieve service account secret
-SECRET=$(kubectl --kubeconfig="${KUBECONFIG}" \
---namespace "${USERNAME}" \
-get serviceaccount "${SERVICE_ACCOUNT}" -o json | python3 -c 'import json,sys;obj=json.load(sys.stdin);print(obj["secrets"][0]["name"])')
-
-if [[ -z "${SECRET// }" ]]; then
-    echo "secret for SA ${SERVICE_ACCOUNT} is not found"
-    exit 1
+# On development setups bypass all the rest of the logic for creating a limited-privs config
+# and just use the admin config
+# Note that this is not secure and should not be used in production
+# Moreover, when testing the security of the spark service on dev, this should be manually disabled
+if [ "$SWAN_DEV" = "true" ]; then
```

Comment:
So moving this down here means that before, dev deployments could not possibly work even if they provided a valid KUBECONFIG because the cern-spark-user chart was not installed?

Reply:
Yes, it was a bit messy. Hopefully it gets better now, at least I could test this in a dev environment.